src/main/java/reactor/netty/transport/ProxyProvider.java

```diff
@@ -155,24 +174,29 @@ public final ProxyHandler newProxyHandler() {
 	}
 
 	/**
-	 * Should proxy the given address
+	 * Returns true when the given {@link SocketAddress} should be reached via the configured proxy. When the method
+	 * returns false, the client should reach the address directly and bypass the proxy
 	 *
 	 * @param address the address to test
 	 * @return true if of type {@link InetSocketAddress} and hostname candidate to proxy
 	 */
 	public boolean shouldProxy(SocketAddress address) {
-		return address instanceof InetSocketAddress && shouldProxy(((InetSocketAddress) address).getHostString());
+		return address instanceof InetSocketAddress && !nonProxyHostPredicate.test(address);
 	}
 
 	/**
-	 * Should proxy the given hostname
+	 * Returns true when the given hostname should be reached via the configured proxy. When the method returns false,
+	 * the client should reach the address directly and bypass the proxy
 	 *
 	 * @param hostName the hostname to test
-	 * @return true if candidate to proxy
+	 * @return true if should be proxied
+	 * @deprecated as of 0.9.10. use {@link #shouldProxy(SocketAddress)}. This method will be removed in version 1.0.0.
 	 */
+	@Deprecated
 	public boolean shouldProxy(@Nullable String hostName) {
-		return nonProxyHosts == null || hostName == null || !nonProxyHosts.matcher(hostName)
-		                                                                  .matches();
+		return nonProxyHostPredicate == ALWAYS_PROXY || hostName == null ||
+				!((RegexShouldProxyPredicate) nonProxyHostPredicate).pattern.matcher(hostName)
```

Comment:
dangerous assumption that this can always be cast to `RegexShouldProxyPredicate`. Couldn't it directly use the `Predicate#test` method instead? (which is more generic)

Reply:
This should not be used by anybody outside of Reactor Netty.
However you are right.
I would suggest here to check only for `instanceof` as before that change there wasn't a possibility for using a Predicate and also as there an explicit recommendation to use the new method.

So this one should work as it was before:

```
		return nonProxyHostPredicate == ALWAYS_PROXY || hostName == null ||
				((nonProxyHostPredicate instanceof RegexShouldProxyPredicate) && !((RegexShouldProxyPredicate) nonProxyHostPredicate).pattern.matcher(hostName))
```

Reply:
The concerns make sense to me - I'll push up the change.